language/module.md

```diff
@@ -0,0 +1,71 @@
+# Puppet Module Schema
+
+This document describes allowed (optional) and required files and directories in
+a Puppet Module.
+
+## What is a Puppet Module
+
+A module in the sense of this document is a compressed tar archive. It is
+usually distributed via [forge.puppet.com](https://forge.puppet.com/). A module
+is usually developed in a git repository. Some files in the git repository are
+only used for development and testing. They shall not be released.
+
+Common files often seen in a vcs repository that are used for development but
+shall not be released:
+
+* `/spec/`
+* `/Rakefile`
+* `/Gemfile`
+* `/.gitignore`
+* `/.github/`
+* `/.devcontainer`
+* `/Dockerfile`
+
+Note that above are just examples and not a complete list. The goal of this
+document is to provide an allowlist, not a denylist.
```

Comment:
If the list of files is just an example, I'd keep it short and inline them. A listing with bullet points draws attention.

Reply:
good idea, updated it. we could also shorten the list.